src/main/java/com/hltech/contracts/judged/publisher/expectations/PactExpectationsReader.java

```diff
@@ -1,16 +1,61 @@
 package com.hltech.contracts.judged.publisher.expectations;
 
 import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
 
-import java.util.HashMap;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Map;
 
+@Slf4j
 public class PactExpectationsReader implements ExpectationsReader {
 
+    public static final String PACTS_LOCATION = "pactsLocation";
+
+    private final String pactsLocation;
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
     public PactExpectationsReader(Map<String, String> configuration) {
+        if (configuration.containsKey(PACTS_LOCATION)) {
+            pactsLocation = configuration.get(PACTS_LOCATION);
+        } else {
+            throw new IllegalStateException("Invalid configuration of PactExpectationsReader. Missing configuration property: " + PACTS_LOCATION);
+        }
     }
 
-    public Map<String, JsonNode> read() {
-        return new HashMap<String, JsonNode>();
+    public List<Expectation> read() {
+        List<Expectation> results = new ArrayList<>();
+        try {
+            File pactsLocationDir = new File(pactsLocation);
+            if (pactsLocationDir.exists()) {
+                File[] files = pactsLocationDir.listFiles();
+                for (File file : files) {
+                    if (file.isFile() && file.getName().endsWith(".json")) {
+                        results.add(new Expectation(getProviderName(file), readContent(file)));
```

Comment:
instead of 
`results.add(new Expectation(getProviderName(file), readContent(file)))` 
try something like this
`results.add(readExpectation(file)))`